test/e2e/source_test.go

```diff
@@ -26,8 +26,11 @@ func TestSource(t *testing.T) {
 	cleanupVcsim := CreateSimulator(t, clients)
 	defer cleanupVcsim()
 
-	// Create a job/svc that listens for events and then quits N seconds after the first is received.
-	name, wait, cancelListener := RunJobListener(t, clients)
+	// Create a job/svc that listens for expectedCount of events of expectedType
+	// It will quit after meeting those criteria, or be cleaned up by cancelListener
+	expectedType := "com.vmware.vsphere.VmPoweredOffEvent.v0"
```

Comment:
I opted for hardcoding it because it seemed like a per test value. If we wrote a different test using the listener, I was thinking we would want to change the execpted event type and count.

Plus we can see that the event we are expecting is related to the actions we describe in the script further down (i.e. turning off vms), maybe making the test more clear?